### PR TITLE
Manpage Make Target

### PR DESCRIPTION
**Pull request by bagasme**
This PR adds make target for manpage, and also stub (empty) markdown source to generate them.

This issue is part of resolving #11399.

The actual manpage content will be provided in next PR when this PR get merged.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -118,6 +118,9 @@ SWAGGER_SPEC_S_TMPL := s|"basePath": *"/api/v1"|"basePath": "{{AppSubUrl}}/api/v
 SWAGGER_SPEC_S_JSON := s|"basePath": *"{{AppSubUrl}}/api/v1"|"basePath": "/api/v1"|g
 SWAGGER_NEWLINE_COMMAND := -e '$$a\'
 
+#Manpage
+MANPAGE := man/man1/gitea.1 man/man5/gitea.app.ini.5
```
Comment (silverwind) [inline Makefile:134]: ~~Would rename variable to `MANPAGE_DIRS`.~~

Edit: Nevermind, I see it relies on the individual entries.

**Comment by silverwind** [comment]
I think you need to run `make vendor` and add the new files in the `vendor` directory.

**Comment by bagasme** [comment]
@silverwind I don't think vendoring will be appropriate for this. It is similar to `misspell` target: if `go-md2man` isn't found, `go get` it. `go-md2man` is executable, not library one.

**Comment by silverwind** [comment]
Ok, I was just wondering about the entry in `go.sum` but I guess as long as it's not in `go.mod`, `go vendor` will not pick it up as a vendoring candidate.

File: .gitignore [hunk 2]
```diff
@@ -94,3 +94,7 @@ prime/
 
 # Make evidence files
 /.make_evidence
+
+# Manpage
+man/man1/gitea.1
```
Comment (6543) [inline .gitignore:111]: why are this files still in this pull diff?

Comment (bagasme) [reply .gitignore:111]: @6543 i think this PR build manpages under `man/` so I had to `.gitignore` resulting files.

**Review by lafriks — APPROVED** [review]

**Comment by silverwind** [comment]
Would prefer this to land in the same PR with the actual content for https://github.com/go-gitea/gitea/issues/11399.

File: Makefile [hunk 3]
```diff
@@ -272,6 +276,16 @@ misspell:
 	fi
 	misspell -w -i unknwon $(GO_SOURCES_OWN)
 
+
+.PHONY: manpage
+manpage:
+	@hash go-md2man > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		$(GO) get -u github.com/cpuguy83/go-md2man; \
```
Comment (silverwind) [inline Makefile]:
```suggestion
		GO111MODULE=off $(GO) get -u github.com/cpuguy83/go-md2man; \
```

this prevents modification of `go.sum` and `go.mod` when running the target.

**Comment by 6543** [comment]
@bagasme  #13429 could be helpfull ...

**Pull request closed without merging**